modules/common/util/template_util.go

```diff
@@ -153,11 +230,16 @@ func lower(s string) string {
 func ExecuteTemplateData(templateData string, data interface{}) (string, error) {
 
 	var buff bytes.Buffer
+	var err error
 	funcs := template.FuncMap{
-		"add":   add,
-		"lower": lower,
+		"add":                      add,
+		"execTempl":                execTempl,
+		"indent":                   indent,
+		"lower":                    lower,
+		"removeNewLines":           removeNewLines,
+		"removeNewLinesInSections": removeNewLinesInSections,
```

Comment:
this is proably the only concern i have with this approach.

this is a nice approach however if we want to go this route
then i think we should also allow passing extra functions via the template class

basically if we go this route i think it would make esne to enable the operator to populate template.ExtaFuncMap with functions defined in the operator.

that just avoid having to always update libcommon first and bumping the libcommon version for functions that may only make sense for a single operator.

that could be a separate PR, to make it possible for client of libcommon to pass in a function map that will be combined with the default ones provided by libcommon.

Reply:
yes we can enhance this and could be used then from within the operators. but if people see the need to have additional functions, we should move them into here and can be re-used by others and not end up each of the operators create its own functions. but for specific use cases it makes sense. Lets do this in a follow up

Reply:
I have created https://issues.redhat.com/browse/OSPRH-10672 to track it

Reply:
ack sound good to me